pulsar-client-api/src/main/java/org/apache/pulsar/client/internal/ReflectionUtils.java

```diff
@@ -51,12 +51,12 @@
         try {
             try {
                 // when the API is loaded in the same classloader as the impl
-                return (Class<T>) DefaultImplementation.class.getClassLoader().loadClass(className);
+                return (Class<T>) Class.forName(className, true, DefaultImplementation.class.getClassLoader());
```

Comment:
Why set `initialize` to `true` here? Will it change the behaviour here since `.loadClass(className)` will not do the initialization.

Reply:
@freeznet in my experience I saw that it is better to fully initialise the classes in environments where there are dynamic classloaders, because you may see weird errors when you shutdown the classloader and you still try to use the class: if the class is not fully loaded you can see NoClassDefFound errors or other link related errors.

I can try to use 'false' and test again my code if you feel strong